tests/components/volvo/conftest.py

```diff
@@ -0,0 +1,232 @@
+"""Define fixtures for Volvo unit tests."""
+
+from collections.abc import AsyncGenerator, Awaitable, Callable, Generator
+from unittest.mock import AsyncMock, patch
+
+from _pytest.fixtures import SubRequest
+import pytest
+from volvocarsapi.api import VolvoCarsApi
+from volvocarsapi.auth import AUTHORIZE_URL, TOKEN_URL
+from volvocarsapi.models import (
+    VolvoCarsAvailableCommand,
+    VolvoCarsLocation,
+    VolvoCarsVehicle,
+)
+from volvocarsapi.scopes import DEFAULT_SCOPES
+from yarl import URL
+
+from homeassistant import config_entries
+from homeassistant.components.application_credentials import (
+    ClientCredential,
+    async_import_client_credential,
+)
+from homeassistant.components.volvo.const import CONF_VIN, DOMAIN
+from homeassistant.const import CONF_API_KEY, CONF_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+from homeassistant.setup import async_setup_component
+
+from .common import async_load_fixture_as_json, async_load_fixture_as_value_field
+from .const import (
+    CLIENT_ID,
+    CLIENT_SECRET,
+    DEFAULT_MODEL,
+    DEFAULT_VIN,
+    MOCK_ACCESS_TOKEN,
+    REDIRECT_URI,
+    SERVER_TOKEN_RESPONSE,
+)
+
+from tests.common import MockConfigEntry
+from tests.test_util.aiohttp import AiohttpClientMocker
+from tests.typing import ClientSessionGenerator
+
+
+def pytest_configure(config: pytest.Config):
+    """Configure pytest."""
+    config.addinivalue_line("markers", "use_model(name): mark test to use given model")
+
+
+def model(name: str):
+    """Define which model to use when running the test. Use as a decorator."""
+    return pytest.mark.use_model(name)
+
+
+@pytest.fixture
+def model_from_marker(full_model_from_marker: str) -> str:  # pylint: disable=hass-argument-type
+    """Get model from marker."""
+    return full_model_from_marker[: full_model_from_marker.index("_")]
+
+
+@pytest.fixture
+def full_model_from_marker(request: SubRequest) -> str:  # pylint: disable=hass-argument-type
+    """Get full model from marker."""
+    marker = request.node.get_closest_marker("use_model")
+    return marker.args[0] if marker is not None else DEFAULT_MODEL
```

Comment:
Interesting approach, however, I think in the end it looks weird
```py
@pytest.mark.parametrize(
    "",
    [
        pytest.param(marks=model("xc40_electric_2024")),
        pytest.param(marks=model("s90_diesel_2018")),
    ],
)
```
I once implemented a nice way to do multiple fixtures, you can check `bosch_alarm` for a simple example, or `smartthings`. There we just have a fixture that returns a `str` which we use, and if that is not overwritten, we use every str that we predefined